32-bit build fails

Hi, I've tried to compile lan-mouse both natively on the target i686-unknown-linux-gnu, and also crosscompiling from x86_64.
Both tries fail with the same error:

> error[E0308]: mismatched types
  --> /home/lars/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lan-mouse-0.8.0/src/frontend/gtk.rs:94:44
   |
94 |             let len = usize::from_be_bytes(len);
   |                       -------------------- ^^^ expected an array with a fixed size of 4 elements, found one with 8 elements
   |                       |
   |                       arguments to this function are incorrect
   |
note: associated function defined here
  --> /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/num/mod.rs:1256:5
   = note: this error originates in the macro `uint_impl` (in Nightly builds, run with -Z macro-backtrace for more info)
For more information about this error, try `rustc --explain E0308`.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,6 +35,7 @@ once_cell = "1.19.0"
 num_enum = "0.7.2"
 hostname = "0.4.0"
 slab = "0.4.9"
+endi = "1.1.0"
 
 [target.'cfg(unix)'.dependencies]
 libc = "0.2.148"

--- a/src/frontend/gtk.rs
+++ b/src/frontend/gtk.rs
@@ -11,6 +11,7 @@ use std::{
 use crate::frontend::{gtk::window::Window, FrontendRequest};
 
 use adw::Application;
+use endi::{Endian, ReadBytes};
 use gtk::{
     gdk::Display, glib::clone, prelude::*, subclass::prelude::ObjectSubclassIsExt, IconTheme,
 };
@@ -85,16 +86,14 @@ fn build_ui(app: &Application) {
     gio::spawn_blocking(move || {
         match loop {
             // read length
-            let mut len = [0u8; 8];
-            match rx.read_exact(&mut len) {
-                Ok(_) => (),
+            let len = match rx.read_u64(Endian::Big) {
+                Ok(l) => l,
                 Err(e) if e.kind() == ErrorKind::UnexpectedEof => break Ok(()),
                 Err(e) => break Err(e),
             };
-            let len = usize::from_be_bytes(len);
 
             // read payload
-            let mut buf = vec![0u8; len];
+            let mut buf = vec![0u8; len as usize];
             match rx.read_exact(&mut buf) {
                 Ok(_) => (),
                 Err(e) if e.kind() == ErrorKind::UnexpectedEof => break Ok(()),

--- a/src/frontend/gtk/window.rs
+++ b/src/frontend/gtk/window.rs
@@ -10,6 +10,7 @@ use std::net::TcpStream;
 
 use adw::prelude::*;
 use adw::subclass::prelude::*;
+use endi::{Endian, WriteBytes};
 use glib::{clone, Object};
 use gtk::{
     gio,
@@ -265,8 +266,7 @@ impl Window {
         let mut stream = self.imp().stream.borrow_mut();
         let stream = stream.as_mut().unwrap();
         let bytes = json.as_bytes();
-        let len = bytes.len().to_be_bytes();
-        if let Err(e) = stream.write(&len) {
+        if let Err(e) = stream.write_u64(Endian::Big, bytes.len() as u64) {
             log::error!("error sending message: {e}");
         };
         if let Err(e) = stream.write(bytes) {

